### PR TITLE
Add delete for PropertyStore in Helix REST

### LINKED ISSUE #1048: Support write functionality for PropertyStore in Helix REST
Support write functionality for PropertyStore

This could be useful for applications that store custom data in PropertyStore.

### PR DESCRIPTION
**Pull request by narendly**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fixes #1048 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This commit adds the delete endpoint for PropertyStore in Helix REST.

### Tests

- [x] The following tests are written for this issue:

testDeletePropertyStore

- [x] The following is the result of the "mvn test" command on the appropriate module:

[INFO] Tests run: 162, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 38.947 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 162, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  45.025 s
[INFO] Finished at: 2020-06-09T15:21:42-07:00

### Commits

- [x] My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation (Optional)

- [ ] In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Code Quality

- [x] My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java [hunk 1]
```diff
@@ -134,4 +135,28 @@ public Response putPropertyByPath(@PathParam("clusterId") String clusterId,
...
+   */
+  @DELETE
+  @Path("{path: .+}")
+  public Response deletePropertyByPath(@PathParam("clusterId") String clusterId,
+      @PathParam("path") String path) {
+    path = "/" + path;
+    if (!ZkValidationUtil.isPathValid(path)) {
+      LOG.info("The propertyStore path {} is invalid for cluster {}", path, clusterId);
+      return badRequest(
+          "Invalid path string. Valid path strings use slash as the directory separator and names the location of ZNode");
```
Comment (pkuwm) [inline helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java:153]: Nit, long line? Maybe split it into two strings concatenation?

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java [hunk 2]
```diff
@@ -134,4 +135,28 @@ public Response putPropertyByPath(@PathParam("clusterId") String clusterId,
...
+    }
+    final String recordPath = PropertyPathBuilder.propertyStore(clusterId) + path;
+    ZkBaseDataAccessor<byte[]> propertyStoreDataAccessor = getByteArrayDataAccessor();
```
Comment (pkuwm) [inline helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java:156]: Nit, `ZkBaseDataAccessor` -> `BaseDataAccessor`?

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java [hunk 3]
```diff
@@ -134,4 +135,28 @@ public Response putPropertyByPath(@PathParam("clusterId") String clusterId,
...
+    if (!propertyStoreDataAccessor.remove(recordPath, AccessOption.PERSISTENT)) {
+      return serverError("Failed to delete PropertyStore record in path: " + path);
+    }
+    return OK();
```
Comment (pkuwm) [inline helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java:160]: Nit, if I were a user, I would expect to see a clear response indicating the request is successful:
```
{
  "success": true
}
```
I think this is included in helix rest improvement plan.

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java (hunk 1 above)
Comment (narendly) [reply helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java:153]: This has been formatted with helix-style, so I don't think we need to do a string concatenation.

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java (hunk 2 above)
Comment (narendly) [reply helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java]: changed!

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java (hunk 1 above)
Comment (pkuwm) [reply helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java:152]: helix-style won't be able to limit such long string within 100 chars per line. We just need to do it manually. If we have checkstyle for 100 chars check, this won't pass.

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java (hunk 3 above)
Comment (narendly) [reply helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java:159]: Let's discuss that as part of the rest improvement plan. We need to discuss what would make the most sense. 

This is done to keep it consistent with other DELETE endpoints.

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java (hunk 1 above)
Comment (narendly) [reply helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java:152]: If it's not something we could strictly enforce, then I don't see a strong reason for doing so. That is if we have a checkstyle check- we don't have this check.

If you want to enforce a lint/checkstyle check, then we could again do it project-wide.

**Pull request merged**